pyteal/ast/gitxn_test.py

```diff
@@ -7,35 +8,75 @@
 
 
 def test_gitxn_invalid():
-    with pytest.raises(TealInputError):
-        GitxnExpr(0, TxnField.sender).__teal__(teal5Options)
-
-    with pytest.raises(TealInputError):
-        Gitxn[MAX_GROUP_SIZE].sender()
-
-    with pytest.raises(TealInputError):
-        Gitxn[-1].asset_sender()
-
-    with pytest.raises(TealInputError):
-        Gitxn[Bytes("first")].sender()
+    for ctor, e in [
+        (
+            lambda: Gitxn[MAX_GROUP_SIZE],
+            TealInputError,
+        ),
+        (
+            lambda: Gitxn[-1],
+            TealInputError,
+        ),
+    ]:
+        with pytest.raises(e):
+            ctor()
 
 
 def test_gitxn_valid():
     for i in range(MAX_GROUP_SIZE):
         Gitxn[i].sender()
 
 
-def test_gitxna_invalid():
-    with pytest.raises(TealInputError):
-        GitxnaExpr("Invalid_type", TxnField.application_args, 1).__teal__(teal6Options)
-
-    with pytest.raises(TealInputError):
-        GitxnaExpr(0, TxnField.application_args, "Invalid_type").__teal__(teal6Options)
+def test_gitxn_expr_invalid():
+    for f, e in [
+        (
+            lambda: GitxnExpr(Int(1), TxnField.sender),
+            TealInputError,
+        ),
+        (
+            lambda: GitxnExpr(1, TxnField.sender).__teal__(teal5Options),
+            TealInputError,
+        ),
+    ]:
+        with pytest.raises(e):
+            f()
+
+
+def test_gitxn_expr_valid():
+    GitxnExpr(1, TxnField.sender).__teal__(teal6Options)
+
+
+def test_gitxna_expr_invalid():
+    for f, e in [
+        (
+            lambda: GitxnaExpr("Invalid_type", TxnField.application_args, 1),
+            TealInputError,
+        ),
+        (
+            lambda: GitxnaExpr("Invalid_type", TxnField.application_args, 1),
+            TealInputError,
```

Comment:
These cases are the same?

Reply:
@jasonpaulos Thanks for catching the duplication.  I fixed the test case via https://github.com/michaeldiamant/pyteal/pull/1/commits/acd8f35b3a754fa80d48b246c58b6229c40e6e7d.